src/counterfactuals/core_struct.jl

```diff
@@ -0,0 +1,155 @@
+"""
+A struct that collects all information relevant to a specific counterfactual explanation for a single individual.
+"""
+mutable struct CounterfactualExplanation <: AbstractCounterfactualExplanation
```

Comment:
imo this could live in the base file

Reply:
This was one of the main concerns I had, I'm not sure which one to go for, couldn't find a specific rule for this kind of thing yet.